src/main/resources/assets/admin/common/js/form/inputtype/text/HtmlAreaCK.ts

```diff
@@ -404,9 +405,37 @@ module api.form.inputtype.text {
             const valueObj: api.data.Value = this.newInitialValue();
             this.setStringValue(valueObj.getPropertySet(), value);
 
+            this.parseReferences(valueObj, value);
+
             this.notifyOccurrenceValueChanged(occurrence, valueObj);
         }
 
+        private parseReferences(valueObj: api.data.Value, value: string) {
+            const contentRegexp = /<a\s+(?:[^>]*?\s+)?href=("|'|\\'|\\")(content:\/\/|media:\/\/download\/)(.+?)\1/g;
+            const imageRegexp = /<img\s+(?:[^>]*?\s+)?src=("|'|\\'|\\")(image:\/\/)(.+?)\1/g;
+
+            let contentMatch = contentRegexp.exec(value);
+            let imageMatch = imageRegexp.exec(value);
+
+            while (contentMatch) {
+                const contentReferenceId = contentMatch.length == 4 ? contentMatch[3] : null;
+
+                if (contentReferenceId) {
+                    valueObj.getPropertySet().addReference('references', Reference.from(new api.content.ContentId(contentReferenceId)));
+                }
+                contentMatch = contentRegexp.exec(value);
+            }
+
+            while (imageMatch) {
+                const imageReferenceId = imageMatch.length == 4 ? imageMatch[3] : null;
+
+                if (imageReferenceId) {
+                    valueObj.getPropertySet().addReference('references', Reference.from(new api.content.ContentId(imageReferenceId)));
+                }
+                imageMatch = imageRegexp.exec(value);
+            }
+        }
+
```

Comment:
We should refactor this. The only difference between extracting image references and other content references is regex that we are using for parsing.
So, make a function called "extractReferences" which will take a regex as an argument and call this function twice from parseReferences, first time passing contentRegexp and second time passing imageRegexp.